### PR TITLE
PR: Update FontAwesome to 5.3.1

### LINKED ISSUE #87: Update FontAwesome to its 5 version
Could you update FA to the last version 5.2.0 ?

### PR DESCRIPTION
**Pull request by darkvertex**
Fixes #87

----

Hello! Here's my contribution of **[FontAwesome 5.1.0](https://fontawesome.com/icons?d=gallery&v=5.1.0&m=free)**.

Since FontAwesome 5.x **breaks compatibility** with 4.x and splits itself into more fonts -- per "style", as they call it -- I kept the "fa" prefix for 4.7 and added **3 new prefixes** for 5.x as:

- `fa5` (regular)
- `fa5s` (solid)
- `fa5b` (brands)

I decided to put in "5", the major version, in the prefix so that in the future when FontAwesome 6.x comes out and potentially breaks backwards glyph compatibility again, we can just add another prefix and call it a day.

I tried to edit as little as possible so I hope all the regular tests are still passing. If not, I will try to fix what I can if I know how.

### CODE REVIEW

**Comment by darkvertex** [comment]
By the way I noticed the Scrutinizer CI does not seem to report "Completed".

It's stuck "Pending" but when clicking through to the job it says it's done and didn't find new issues.

Not sure what I can do about that, if anything.

**Comment by darkvertex** [comment]
I don't know what's wrong with the Appveyor build. It quit abruptly without any errors?

**Comment by darkvertex** [comment]
I've noticed icons that should be solid render as regular. I suspect there is a small bug with how the fonts are being loaded and the solid is not getting taken, yet its icon codes and prefix is registered properly.

Any idea where the issue may be?

I have this simple example file:
```python
# -*- coding: utf-8 -*-

# Standard library imports
import sys

# Third party imports
from qtpy import QtCore, QtWidgets
from six import unichr

# Local imports
import qtawesome as qta


class AwesomeExample(QtWidgets.QDialog):

    def __init__(self):
        super(AwesomeExample, self).__init__()

        # Get icons by name.
        fa_icon = qta.icon('fa5s.flag')
        fa_button = QtWidgets.QPushButton(fa_icon, 'Font Awesome!')

        # Render a label with this font
        label1 = QtWidgets.QLabel(unichr(0xf118) + ' ' + 'Regular')
        label1.setFont(qta.font('fa5', 16))
        label2 = QtWidgets.QLabel(unichr(0xf118) + ' ' + 'Solid')
        label2.setFont(qta.font('fa5s', 16))

        # Layout
        vbox = QtWidgets.QVBoxLayout()
        widgets = [fa_button, label1, label2]

        for w in widgets:
            vbox.addWidget(w)

        self.setLayout(vbox)
        self.setWindowTitle('Awesome')
        self.show()


def main():
    app = QtWidgets.QApplication(sys.argv)
    QtCore.QTimer.singleShot(10000, app.exit)
    _ = AwesomeExample()
    sys.exit(app.exec_())


if __name__ == '__main__':
    main()
```
which renders as:
image

**Comment by darkvertex** [comment]
About the failing AppVeyor builds, [it seems to fail specifically with Python 3.5](https://ci.appveyor.com/project/spyder-ide/qtawesome/build/1.0.73) and not with 3.4 or earlier.

I noticed your master branch's `setup.py` says:
```
        # Supported Python versions
        'Programming Language :: Python :: 2',
        'Programming Language :: Python :: 2.6',
        'Programming Language :: Python :: 2.7',
        'Programming Language :: Python :: 3',
        'Programming Language :: Python :: 3.2',
        'Programming Language :: Python :: 3.3',
        'Programming Language :: Python :: 3.4',]
```
No Python 3.5 there... so why does `appveyor.yml` try to test 3.5 in [line 19](https://github.com/spyder-ide/qtawesome/blob/master/appveyor.yml#L19) and [line 26](https://github.com/spyder-ide/qtawesome/blob/master/appveyor.yml#L26)?

Want me to make a PR to remove 3.5 from the testing matrix?

**Comment by doudz** [comment]
Looks like fa5 overwrite fa5s, I'll give a look

**Comment by doudz** [comment]
I have proposed a small fix

**Comment by darkvertex** [comment]
@ccordoba12 Shall I rebase over `master` to get the fixed testing infra? Is that okay or do you prefer I merge the branch into this one?

**Comment by ccordoba12** [comment]
I think a rebase is easier in this case.

**Comment by darkvertex** [comment]
Just rebased and in doing so, squashed some commits for clarity of history.

Looks like the circleci tests succeeded and this time, so did AppVeyor's! :tada:

I'll look into upgrading this from 5.1.0 to 5.2.0 then. Just have to update the ttfs, md5 hashes and dump the icon metadata again.

**Comment by ccordoba12** [comment]
@darkvertex, this is looking really great, thanks!

However, you need to a shim between the icon names in FontAwesome 4 with the ones in FontAwesome 5. Else Spyder will stop to work.

You can find that work in PR #82.

**Comment by darkvertex** [comment]
Okay, I'll look at sorting out those shims while updating to 5.2.0

**Comment by darkvertex** [comment]
I rebased to fix the little conflicts from the other PRs that got merged before this one.

I also made the changes to update FA from 5.1.0 to 5.3.1, the very latest and greatest. :man_dancing:

As for the shims, I was wondering if they are truly necessary when you consider that I am not affecting the "fa" namespace at all? As per the PR in its current form, "fa" remains as FontAwesome 4.7 forever. My understanding is that the shims information is for keeping the namespace, using the same font and aliasing the changes from old to the new, but since I'm introducing `fa5`, `fa5s` and `fa5b` yet leaving `fa` as-is, is it still relevant? (Correct me if I'm wrong.)

I've never used Spyder IDE at all so forgive my poor understanding of its inner workings.

**Comment by darkvertex** [comment]
If you would rather I use the shims so as to update the old namespace in-place while introducing the new "solid" and "brands" variants, I could change things like so:

`fa5` -> `fa`
`fa5s` -> `fas`
`fa5b` -> `fab`

Would you prefer that?

**Comment by darkvertex** [comment]
Also, I checked the other PR with the `shims.py` https://github.com/spyder-ide/qtawesome/blob/614ccbab9b89cba61ed6ab69b2a2bb442cc2baa8/update/shims.py
but I don't understand what I'm supposed to feed `update_fa_dict()`.

Docstring reads:
```
:param namedict: dictionary with aliases for icons, like Spyder's icon_manager.py
```
but where do I get that dict from?

**Comment by ccordoba12** [comment]
> As for the shims, I was wondering if they are truly necessary when you consider that I am not affecting the "fa" namespace at all? As per the PR in its current form, "fa" remains as FontAwesome 4.7 forever. My understanding is that the shims information is for keeping the namespace, using the same font and aliasing the changes from old to the new, but since I'm introducing fa5, fa5s and fa5b yet leaving fa as-is, is it still relevant?

If FontAwesome 5 has its own namespace, then I think that's much better. That way Spyder won't loose its current look and feel, and other projects use icons from any version they want.

File: README.md [hunk 1]
```diff
@@ -51,7 +51,7 @@ import qtawesome as qta
 
 ```python
 # Get icons by name.
-fa_icon = qta.icon('fa.flag')
+fa_icon = qta.icon('fa5s.flag')
```
Comment (ccordoba12) [inline README.md]: Why did you change everything to use `fa5s`? I'd prefer for you to use `fa5`.

Comment (darkvertex) [reply README.md]: It was for visual consistency with your existing screenshot:
https://raw.githubusercontent.com/spyder-ide/qtawesome/master/qtawesome-screenshot.gif

In your README you use `fa.flag` which for FA 4.x is a solid looking flag:
https://fontawesome.com/v4.7.0/icon/flag
but then in FA 5.x, `fa5.flag` is the "regular" flag, which is rendered as an outline style:
https://fontawesome.com/icons/flag?style=regular
and the solid one got put in the "solid" style, which is `fa5s.flag`:
https://fontawesome.com/icons/flag?style=solid

This way, the icons in your example don't change visually and are only updated to the "modern" namespace of fa5s.

Do you still want me to use `fa5.flag` in all the examples?

File: qtawesome/__init__.py [hunk 2]
```diff
@@ -55,22 +66,28 @@ def icon(*names, **kwargs):
     The ``prefix`` corresponds to the font to be used and ``name`` is the
     name of the icon.
 
-     - The prefix corresponding to Font-Awesome is 'fa'
+     - The prefix corresponding to Font-Awesome 4.x is 'fa'
+     - The prefix corresponding to Font-Awesome 5.x (regular) is 'fa5'
+     - The prefix corresponding to Font-Awesome 5.x (solid) is 'fa5s'
+     - The prefix corresponding to Font-Awesome 5.x (brands) is 'fa5b'
```
[the remark this answers is not in the document]
Comment (darkvertex) [reply qtawesome/__init__.py]: Sure thing. You mean somewhere in the `README.md` I presume?

[the remark this answers is not in the document]
Comment (darkvertex) [reply qtawesome/__init__.py]: I just improved the README a little bit. What do you think?

**Comment by darkvertex** [comment]
Have you had a chance to look at my recent changes, @ccordoba12? Does it look good to you now?

File: qtawesome/__init__.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ccordoba12) [reply qtawesome/__init__.py]: It looks, thanks!

[the remark this answers is not in the document]
Comment (ccordoba12) [reply qtawesome/__init__.py]: Please also add the same improvements to our documentation inside the `docs` folder.

[the remark this answers is not in the document]
Comment (darkvertex) [reply qtawesome/__init__.py]: Will do.

[the remark this answers is not in the document]
Comment (darkvertex) [reply qtawesome/__init__.py]: All done!

**Comment by darkvertex** [comment]
I updated `/docs` to match `README.md` improvements.

I think it's all good to go.

**Comment by darkvertex** [comment]
Thank you! I'm really happy to see this get merged at last.

When do you plan to release a new version to PyPi? I noticed you had tagged a bunch of PRs (including this one) with the v0.5 milestone:
https://github.com/spyder-ide/qtawesome/milestone/2

**Comment by ccordoba12** [comment]
Hopefully today, or tomorrow at most.

**Comment by ccordoba12** [comment]
0.5 is out!

**Comment by darkvertex** [comment]
Amazing!! 😃
